### PR TITLE
Articles API: state=all by itself results in a 500 error

### LINKED ISSUE #4522: Articles API: state=all by itself results in a 500 error
**Describe the bug**

Calling `curl https://dev.to/api/articles?state=all` results in a 500 error.

Although, according to the documentation, that specific parameter is only to be used in conjunction with `username`, used by itself shouldn't result in a 500 error, instead it should return an empty result set.

**To Reproduce**

1. Type `curl https://dev.to/api/articles?state=all` or `curl http://localhost:3000/api/articles?state=all` in your local development environment
2. see error

**Expected behavior**

Calling `curl https://dev.to/api/articles?state=all` should return an empty JSON array

**Additional context**

The issue is likely the missing `else` condition here:

https://github.com/thepracticaldev/dev.to/blob/0d2506f9ef79a32f7474001337161314d1062134/app/services/article_api_index_service.rb#L77-L85

### PR DESCRIPTION
**Pull request by sabunt**
#4522 

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Documentation Update

## Description
Add try for decorate if article nor present 
## Related Tickets & Documents

Fixes #4522 

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by sabunt** [comment]
@rhymes. I think my last version improved response.  Added logic and add rspec. State=all should response  base articles if params state not valid return []

File: app/services/article_api_index_service.rb [hunk 1]
```diff
@@ -81,6 +82,8 @@ def state_articles(state)
     elsif state == "rising"
       Article.published.
         where("positive_reactions_count > ? AND positive_reactions_count < ? AND featured_number > ?", 19, 33, 3.days.ago.to_i)
+    elsif state == "all"
+      base_articles
```
Comment (rhymes) [inline app/services/article_api_index_service.rb]: This changes the API, I think we should stick with the current behavior, even though could be counter intuitive right now.

I would remove this

File: app/services/article_api_index_service.rb [hunk 2]
```diff
@@ -24,8 +24,9 @@ def get
                else
                  base_articles
                end
+    articles.presence || []
```
Comment (rhymes) [inline app/services/article_api_index_service.rb]: this is not needed

**Comment by sabunt** [comment]
@rhymes revert last commit

**Comment by rhymes** [comment]
@sabunt thanks! Please remember not to force push after reviews, unless you're rebasing from master :)

**Comment by sabunt** [comment]
remove conflicts

**Pull request merged**